### PR TITLE
fix(iot-device): Update parameter to send to MQTT connect packet for …

### PR DESCRIPTION
**Pull request by vinagesh**
…a pnp device

 

# Checklist
- [ ] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-java/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [ ] This pull-request is submitted against the `master` branch. 

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem
Updated parameter to send in MQTT connect packet to pnp service to recognize the device as pnp.
Tested this locally using a sample on the hub provided by service team.

### CODE REVIEW

File: device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/mqtt/MqttIotHubConnection.java [hunk 1]
```diff
@@ -42,7 +42,7 @@
     private static final String SSL_PREFIX = "ssl://";
     private static final String SSL_PORT_SUFFIX = ":8883";
 
-    private static final String ModelIdParam = "digital-twin-model-id";
+    private static final String ModelIdParam = "model-id";
```
Comment (prmathur-microsoft) [inline device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/mqtt/MqttIotHubConnection.java:46]: Will there be separate PR for Service Client?

Comment (vinagesh) [reply device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/mqtt/MqttIotHubConnection.java:46]: There is already a PR for it.

Comment (prmathur-microsoft) [reply device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/mqtt/MqttIotHubConnection.java:46]: nm. Just saw this https://github.com/Azure/azure-iot-sdk-java/pull/834/files
